internal/states/encryption/encryption.go

```diff
@@ -0,0 +1,263 @@
+package encryption
+
+import (
+	"fmt"
+	"github.com/hashicorp/go-hclog"
+	"github.com/opentofu/opentofu/internal/states/encryption/encryptionconfig"
+	"github.com/opentofu/opentofu/internal/states/encryption/encryptionflow"
+	"github.com/opentofu/opentofu/internal/tfdiags"
+	"sync"
+)
+
+// New creates a new Encryption object. You can use this object to feed in encryption configuration and then create
+// an encryptionflow.StateFlow or encryptionflow.PlanFlow as needed.
+//
+// Note: due to a lot of procedural code you probably want to use GetSingleton() to obtain the Encryption implementation
```

Comment:
I don't understand this note and what is meant with procedural.

Reply:
Tofu has a bunch of procedural code so the singleton serves as a global scope. I guess this could do with rewriting and better explanation.

Reply:
Yeah, a slight clarification could be of use here.